TPLsList.cmake

```diff
@@ -174,6 +174,7 @@ TRIBITS_REPOSITORY_DEFINE_TPLS(
   MAGMASparse     "cmake/TPLs/"    EX
   Check           "cmake/TPLs/"    EX
   SARMA           "cmake/TPLs/"    EX
+  CDT             "cmake/TPLs/"    EX
```

Comment:
@ldh4, will this be required or optional TPL for STK?  With the TPL currently being declared `EX` it will be automatically hard disabled and will take out anything downstream that has a required dependency on it.

Reply:
This can be an optional TPL. It is only required for one of STK modules, and we are okay having that module disabled if the user doesn't manually enable CDT and provide its installation path.